content/blog/case-weights/index.Rmd

```diff
@@ -0,0 +1,310 @@
+
+```{r sim-wts}
+training_sim <-
+  training_sim %>% 
+  mutate(
+    case_wts = ifelse(class == "class_1", 60, 1),
+    case_wts = importance_weights(case_wts)
+  )
+```
+
+We strongly advise that users set the case weight column before any other tidymodels functions are used. This ensures that they are handled correctly in the analyses that follow. 
+
+Let's use 10-fold cross-validation to resample the data. This case is unaffected by the presence of weights: 
+
+```{r sim-folds}
+set.seed(2)
+sim_folds <- vfold_cv(training_sim, strata = class)
+```
+
+We'll fit a regularized logistic regression model to the data using glmnet: 
+
+```{r sim-model}
+lr_spec <- 
+  logistic_reg(penalty = tune(), mixture = 1) %>% 
+  set_engine("glmnet")
+```
+
+For this model, we need to ensure that the predictors are in the same units. We'll use a recipe to center and scale the data and also add some spline terms for predictors that appear to have a nonlinear relationship with the outcome: 
+
+```{r sim-rec}
+sim_rec <- 
+  recipe(class ~ ., data = training_sim) %>% 
+  step_ns(starts_with("non_linear"), deg_free = 10) %>% 
+  step_normalize(all_numeric_predictors())
+sim_rec
+```
+
+There are a few things to point out here. The recipe automatically detects the case weights even though they are captured by the dot in the right-hand side of the formula. The recipe automatically sets their role and will error if that column is changed in any way. 
+
+As mentioned above, any unsupervised steps are unaffected by importance weights so neither `step_ns()` or `step_normalize()` use the weights in their calculations. 
+
+When using case weights, we would like to encourage users to keep their model and preprocessing tool within a workflow. The workflows package now has a `add_case_weights()` function to help here: 
+
+```{r sim-wflow}
+lr_wflow <- 
+  workflow() %>% 
+  add_model(lr_spec) %>% 
+  add_recipe(sim_rec) %>% 
+  add_case_weights(case_wts)
+lr_wflow
+```
+
+From here, let's tune the model. This code doesn't have to change at all. Because we set the case weights column with `add_case_weights()`, tune and workflows automatically know how to pass the case weights on to the underlying parsnip model. We will add some additional metrics to show the affect of the weights though: 
+
+```{r sim-tune, fig.height=4, fig.width=7, out.width="100%", dev='svg'}
+cls_metrics <- metric_set(sensitivity, specificity)
+
+grid <- tibble(penalty = 10^seq(-3, 0, length.out = 20))
+
+set.seed(3)
+lr_res <- 
+  lr_wflow %>% 
+  tune_grid(resamples = sim_folds, grid = grid, metrics = cls_metrics)
+
+autoplot(lr_res)
+```
+
+Recall that tidymodels assumes that the first level of the outcome factor is the event of interest. Since the first level of the outcome has the fewest values, we would expect that the sensitivity of the model to suffer. These results suggest that the weights are making the model focus on the majority class. 
```

Comment:
```suggestion
Recall that tidymodels assumes that the first level of the outcome factor is the event of interest. Since the first level of the outcome has the fewest values, we would expect the sensitivity of the model to suffer. These results suggest that the weights are making the model focus on the majority class. 
```